src/onnx/parse_slice.cpp

```diff
@@ -77,7 +78,37 @@ struct parse_slice : op_parser<parse_slice>
             op.axes = axes;
         }
 
-        return info.add_instruction(op, args[0]);
+        std::vector<int64_t> raxes;
+        if(std::any_of(steps.begin(), steps.end(), [](auto s) { return s < 0; }))
+        {
+            for(auto i : op.axes)
+            {
+                if(steps[i] < 0)
+                    raxes.push_back(op.axes[i]); // populate negative only axis indices
+            }
+            
+            auto lens = args[0]->get_shape().lens();
+            for(auto axis : raxes)
+            {
+                op.starts[axis] += 1;
+                if(op.starts[axis] == 0)
+                {
+                    op.starts[axis] = INT_MAX;
+                }
+                op.ends[axis] += 1;
+                std::swap(op.starts[axis], op.ends[axis]);
+            }
+        }
+
+        
+
+        auto ins = info.add_instruction(op, args[0]);
+        if(not raxes.empty())
+        {
+            return info.add_instruction(make_op("reverse", {{"axes", raxes}}), ins);
+        }
```

Comment:
Really small, but maybe remove the braces so its consistent with the else branch or add braces to the else.

Reply:
Removed the braces, looks cleaner!